### PR TITLE
Feature - move pil licence number to profile

### PR DESCRIPTION
**Pull request by joefitter**
* updated billing to search pil licence no from profile
* updated pil get/:id to include licence number from profile

### CODE REVIEW

**Comment by joefitter** [comment]
WIP until https://github.com/UKHomeOffice/asl-schema/pull/358

**Comment by lennym** [comment]
This will need the PIL list endpoints modifying in the same way.

e.g. https://internal-ui.dev.asl.homeoffice.gov.uk/establishments/8201/pils

**Review by lennym — DISMISSED** [review]

File: lib/routers/profile/pil.js [hunk 1]
```diff
@@ -98,6 +98,11 @@ router.param('pilId', (req, res, next, id) => {
   Promise.resolve()
     .then(() => {
       return PIL[queryType]().findById(id)
+        .select([
+          'pils.*',
+          'profile.pilLicenceNumber as licenceNumber'
+        ])
```
Comment (lennym) [inline lib/routers/profile/pil.js:104]: We probably need to handle this _a little_ more cleverly for the cases where the user has multiple old PIL records that do not have the same licence number.

i.e. load the profile licence number and the PIL licence number independently, and then if they differ respond with the one from the PIL.

Comment (joefitter) [reply lib/routers/profile/pil.js:104]: good shout, I hadn't thought of that

**Comment by joefitter** [comment]
and yes, the list function comes from schema and has been updated there

File: lib/routers/profile/pil.js [hunk 2]
```diff
@@ -86,34 +86,38 @@ const attachEstablishmentDetails = (req, res, next) => {
...
+    });
+
+  if (!pil) {
+    return next(new NotFoundError());
+  }
+
+  const profile = await Profile.query().findById(pil.profileId);
+
+  if (!pil.licenceNumber || pil.licenceNumber === profile.pilLicenceNumber) {
+    pil.licenceNumber = profile.pilLicenceNumber;
```
Comment (wheelsandcogs) [inline lib/routers/profile/pil.js:113]: do we need the 2nd condition here? Isn't it just a noop?

Comment (joefitter) [reply lib/routers/profile/pil.js:113]: no, this is addressing Lenny's earlier comment: We always want to use the licence number from the profile, apart from if it is different (if you are looking at a legacy migrated pil)

File: lib/routers/profile/pil.js [hunk 3]
```diff
@@ -86,34 +86,38 @@ const attachEstablishmentDetails = (req, res, next) => {
...
-    .catch(next);
+    });
+
+  if (!pil) {
+    return next(new NotFoundError());
+  }
+
+  const profile = await Profile.query().findById(pil.profileId);
+
+  if (!pil.licenceNumber || pil.licenceNumber === profile.pilLicenceNumber) {
```
Comment (joefitter) [inline lib/routers/profile/pil.js:112]:
```suggestion
  if (!pil.licenceNumber) {
```

**Pull request merged**